mono/utils/mono-threads-mach-helper.c

```diff
@@ -24,6 +24,12 @@
 */
 void mono_threads_init_dead_letter (void);
 void mono_threads_install_dead_letter (void);
+
+// mono_thread_info_detach is MONO_API elsewhere, so needs to at least be extern "C" here.
```

Comment:
Just `#include <glib.h>`

Reply:
Relucant, but  ok.

Reply:
Since it's MONO_API, why not simply putting `MONO_API void mono_thread_info_detach (void);`?

Reply:
This part of the code is really cautious in what it includes but I can try adding publib/compiler if you insist. Recall you had to push me even to add glib.h and I was reluctant to do that.

Reply:
And I agree what I have is dubious -- like, we declare two different ways, what is the result?